JITM Package: Update the WooCommerce logo in JITMs (display after Feb 4).

elliottprogrammer opened the pull request:
In this PR we are updating the WooCommerce logo shown in JITM's.
The new logo should **_not_** display **before Feb 4, 2025**.

**Background:**  Slack conversation: p1737629667392579-slack-C01264051NE

#### Screenshots:

BEFORE Feb 4th 2025 | On or After Feb 4th, 2025
--- | ---
![Screen Shot 2025-01-25 at 16 12 00](https://github.com/user-attachments/assets/95141283-79fd-4ae4-862d-60b684c2c363) | ![Screen Shot 2025-01-25 at 16 00 52](https://github.com/user-attachments/assets/4af23134-0c70-4b2a-8c8e-47e32171d752)


<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* If current date is on or **after Feb 4, 2025**, display the New WooCommerce logo svg in Woo JITM's.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
Slack conversation: p1737629667392579-slack-C01264051NE

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Apply this PR branch to a Jurassic Ninja site via the Jetpack Beta plugin, OR to your local Jetpack dev environment  site (using Docker).
* If wpcom PR 171101-ghe-Automattic/wpcom is not shipped yet, you will need to apply the wpcom PR branch to your sandbox and then `define( 'JETPACK__SANDBOX_DOMAIN', 'Your sandbox domain' ); to your Jetpack site (in wp-config.php).
* Even if the above wpcom PR is already shipped, still it is advised that you point your site to your wpcom sandbox anyway, that way you can add these filters (below) ⤵️ to your `0-sandbox.php` file, which makes things easier when trying to test & display specific JITMS.

```php
// Disable JITM cache during development
add_filter( 'jetpack_just_in_time_msg_cache', '__return_false', 1 );
// Disable JITM throttling
add_filter( 'jetpack_jitm_should_return_jitms', '__return_true', 1 );
```
- Make sure your Jetpack site does Not have a paid plan.
- Make sure Jetpack is connected (both site & user).
- Install and activate the WooCommerce plugin.
- Go to the "Products" (WooCommerce-->Products(edit products)) page: (`/wp-admin/edit.php?post_type=product`).
- In the URL add this url query param: `&test_jitm=1` and re-load the page.
- The Woo Test Jitm should now show at the top of the page.
- If the current date is before Feb 4th, the old, classic Woo logo should be showing in the JITM.
- To view/test the new Woo logo, Add this filter to your site's `wp-config.php` file:
```php
add_filter( 'jetpack_jitm_use_new_woo_logo', '__return_true', 1 );
```
- Reload the page and you should now see the new Woo logo svg in the JITM.
- Examine the code change and logic.  See that the new logo will display when the current date is on or after Feb 4th, 2025.

elliottprogrammer committed:
Update the WooCommerce logo in JITMs (display after Feb 4, 2025).

elliottprogrammer committed:
changelog

elliottprogrammer committed:
Fix DateTime & apply filter hook for showing old/new logo.

jeherve commented on projects/packages/jitm/src/class-jitm.php:
```diff
@@ -255,21 +255,47 @@ public function ajax_message() {
-			</style><g><defs><polygon id="SVGID_1_" points="83.8 100 0 100 0 0.3 83.8 0.3 167.6 0.3 167.6 100 "/></defs><clipPath id="SVGID_2_"><use xlink:href="#SVGID_1_" overflow="visible"/></clipPath><g class="st0"><g><defs><rect id="SVGID_3_" width="168" height="100"/></defs><clipPath id="SVGID_4_"><use xlink:href="#SVGID_3_" overflow="visible"/></clipPath><g class="st1"><defs><path id="SVGID_5_" d="M ...
+				// Don't show the new Woo svg logo until after Feb 4th, 2025
+				// After Feb 4th 2025, we can remove this date condition check ( & filter) and the old svg logo.
+				$content_icon = apply_filters( 'woo_jitm_use_new_logo', $date_now >= $feb_4_date )
```
Could you add a docblock to that filter, so our codex can pick it up?

I would also recommend using a `jetpack_jitm` prefix, to match the rest of the package:
https://github.com/Automattic/jetpack/blob/58bed9516439cfa6fd4f6921f914c26137acfa54/projects/packages/jitm/src/class-post-connection-jitm.php#L348

For the `@since` value, you can use `$$next-version$$` as the value, it will be automatically replaced on release day:
https://github.com/Automattic/jetpack/blob/d3f58235992e84fcc8a4b1bc191c56ba6edf213a/projects/packages/README.md#L114-L123

jeherve commented on projects/packages/jitm/src/class-jitm.php:
```diff
@@ -255,21 +255,47 @@ public function ajax_message() {
 	 * @param bool   $full_jp_logo_exists Is there a big JP logo already displayed on this screen.
 	 */
 	public function generate_icon( $content_icon, $full_jp_logo_exists ) {
+		$date_now   = new \DateTime();
+		$feb_4_date = new \DateTime( '02-04-2025' );
```
Would it be worth adding `wp_timezone()` when getting the dates, so as to switch at the right moment regardless of your site settings?

elliottprogrammer replied:
Thanks for the suggestion @jeherve! Yes sure, why not! It wouldn't hurt!
I mean AFAIK I don't think it's completely necessary for the logo to switch for everybody at the exact same time, but then again, sure why not? I think it would be even better that way! 😉

But would you mind checking/reviewing my code changes using `wp_timezone()`?
I don't think I've ever used `wp_timezone()` before, so I'm not completely sure I did it correctly..., Does it look OK to you?

KokkieH replied:
> AFAIK I don't think it's completely necessary for the logo to switch for everybody at the exact same time

Can confirm. As long as it happens at some point on the 4th :)

jeherve replied:
> would you mind checking/reviewing my code changes using wp_timezone()?

You should be able to pass the timezone directly as you build the date object:

`$feb_4_date = new \DateTime( '02-04-2025', wp_timezone() );`

elliottprogrammer committed:
Use 'jetpack_jitm' prefix and add docblock on use_new_woo_logo filter.

elliottprogrammer committed:
Utilize wp_timezone when checking the Feb 4th date.

elliottprogrammer commented:
@jeherve, Thanks for the review, much appreciated! 😃

> Could you add a docblock to that filter, so our codex can pick it up?
>
> I would also recommend using a jetpack_jitm prefix, to match the rest of the package:

Yes sounds good! Thanks for the feedback!
- Added a docblock to the filter
- Prefixed the filter name with `jetpack_jitm_` to match the rest of the package.
- Added the use of `wp_timezone()` when checking for the Feb 4th date. (Can you take a look?)

> For the @since value, you can use $$next-version$$ as the value, it will be automatically replaced on release day:

Oh that's awesome, I didn't know about $$next-version$$!  TIL!! 🙌

elliottprogrammer committed:
Merge branch 'trunk' into update/woo-jitm-logo

elliottprogrammer committed:
Remove use of 'wp_timezone()'.

elliottprogrammer committed:
re-add 'wp_timezone()' and mock in php unit test.

grzegorz-cp reviewed (approved):
Looks good. I was able to see the JITM a few times but not constantly, so this might be an issue with my testing.
